packages/ios_platform_images/ios/Classes/UIImageIosPlatformImages.swift

```diff
@@ -22,6 +22,13 @@
 ///
 /// Note: We don't yet support images from package dependencies (ex.
 /// `AssetImage('icons/heart.png', package: 'my_icons')`).
-+ (UIImage *)flutterImageWithName:(NSString *)name;
 
-@end
+func flutterImage(withName imageName: String) -> UIImage? {
+    for scale in stride(from: Int(UIScreen.main.scale), through: 1, by: -1) {
+      let scaledImage = imageName.replacingOccurrences(of: ".", with: "@\(scale)x.")
+      return UIImage(named: scaledImage, in: Bundle.main, compatibleWith: nil)
+    }
+
+    return UIImage(named: imageName, in: Bundle.main, compatibleWith: nil)
+  }
```

Comment:
The indentation is a bit off in this file. Can you run the formatter locally?
Refer to [this wiki section](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#swift-migration-for-1p-plugins) on installing the formatter on your local machine.

Reply:
@hellohuanlin @stuartmorgan  I have a question about the code I migrated, specifically about the image. Reading about the documentation [UIImage](https://developer.apple.com/documentation/uikit/uiimage) , I imagined that UIImage itself could take care of the image scale.
After @stuartmorgan  comment I came back with the previous code. Below is the implementation but it doesn't work as expected, I'm probably doing something wrong. I don't know what to do in this situation.

func flutterImage(withName name: String) -> UIImage? {
  let components = name.components(separatedBy: "/")
  let filename = components.last
  let path = components.dropLast().joined(separator: "/")

  for screenScale in stride(from: Int(UIScreen.main.scale), to: 1, by: -1) {
    let key = FlutterDartProject.lookupKey(forAsset: "\(path)\(screenScale).0x/\(filename!)")
    if let image = UIImage(named: key, in: Bundle.main, compatibleWith: nil) {
      return image
    }
  }

  let key = FlutterDartProject.lookupKey(forAsset: name)
  return UIImage(named: key, in: Bundle.main, compatibleWith: nil)
}

Reply:
Can you set a breakpoint there and see if Swift has the same `key` value as the ObjC?

Reply:
Oh you are missing a `/` in the path

Reply:
Can you add a unit test there? It would've been an easy-to-find mistake if we had unit test before.

Reply:
> Oh you are missing a `/` in the path: `"\(path)\(screenScale).0x/\(filename!)"`

I removed it because it was //

Reply:
> Can you set a breakpoint there and see if Swift has the same `key` value as the ObjC?
on scale -> key	String	"Frameworks/App.framework/flutter_assets/2.0x/flutter"	
normal ->    key	String	"Frameworks/App.framework/flutter_assets/flutter"

Reply:
@hellohuanlin I get key of ObjC an give me :` key	__NSCFString *	@"Frameworks/App.framework/flutter_assets/flutter"	0x0000600001430230`

Reply:
I meant, the ObjC version has a `/` between path and screenScale: 

`[NSString stringWithFormat:@"%@/%d.0x/%@", path, screenScale, filename]]`

and the Swift version is missing this `/`: 

`"\(path)\(screenScale).0x/\(filename!)"`

Reply:
@hellohuanlin Ok, but it was supposed to load the default image for this reason but it didn't happen.
The image scaled should be `Frameworks/App.framework/flutter_assets//2.0x/flutter`?

Reply:
No there shouldn't be `//` (double slash).

Reply:
Can you set a break point there, take a screenshot of Xcode window (including your code and the debug window). I can help to take a look.

Reply:
![Captura de Tela 2023-09-13 às 20 52 12](https://github.com/flutter/packages/assets/50643541/ab6366d6-e868-4c12-93ee-024a5eff6176)

Reply:
>   let components = name.components(separatedBy: "/")
>   let filename = components.last
>   let path = components.dropLast().joined(separator: "/")

Is there some reason you rewrote this logic?

Reply:
Before I was writing it as if it was ObjC, but now I can write it more clearly.

Reply:
From the screenshot, you added the `/`, and the path seems to be correct. Is there still an issue?

Reply:
Yes, returns nil

Reply:
> Before I was writing it as if it was ObjC, but now I can write it more clearly. Not committed yet.

I'm not sure what this means, but what you have in both the PR and the screenshot is different logic than the working Obj-C version; rather than converting the working logic to Swift, you've written different code. As I explained above logic changes aren't something we want in a conversion PR. The fact that your version doesn't work is an example of why changing the logic while porting to Swift problematic (especially since in cases like the Dart argument parsing you seem to have changed the logic without a clear understanding of what the original code was doing).

Reply:
> Yes, returns nil.

Does the objc version have exactly the same arguments passed in? And what does it return?

Reply:
@hellohuanlin I wrote another ObjC test and the scaled key is:
`key	__NSCFString *	@"Frameworks/App.framework/flutter_assets//2.0x/flutter"	0x00006000018a0c80`, with double slash

Reply:
> > Before I was writing it as if it was ObjC, but now I can write it more clearly. Not committed yet.
> 
> I'm not sure what this means, but what you have in both the PR and the screenshot is different logic than the working Obj-C version; rather than converting the working logic to Swift, you've written different code. As I explained above logic changes aren't something we want in a conversion PR. The fact that your version doesn't work is an example of why changing the logic while porting to Swift problematic (especially since in cases like the Dart argument parsing you seem to have changed the logic without a clear understanding of what the original code was doing).

I came back with the original logic to test, but it still doesn't work. I think it must have some connection with the UIImage, I still haven't figured out why.
In this recent logic that I sent, I cast String to NSString to have the same result as I have in ObjC

Reply:
Hmmm, I'm surprised that ObjC path uses double slash and it still works. I may need to look into this plugin for more details, but as Stuart said, it's better not to modify any behavior in a migration PR. 

To reduce risk of regression, we should backfill some unit tests. For example, adding tests for this function is a good starting point, since we are already making a mistake here. 

Backfilling tests also helps to better understand the existing ObjC behavior, which actually makes migration easier for us.

Reply:
> To reduce risk of regression, we should backfill some unit tests. Adding tests for this function is a good starting point, since we are already making a mistake here.
> 
> Backfilling tests also helps to better understand the existing ObjC behavior, which actually makes migration easier for us.

See https://github.com/flutter/packages/pull/4847#discussion_r1326012844

Reply:
@hellohuanlin In my free time I will investigate the behavior of the ObjC code and compare it with the Swift code.

Reply:
> Hmmm, I'm surprised that ObjC path uses double slash and it still works.

POSIX requires multiple consecutive separators in a path to be treated the same as a single separator, so it's not really surprising.

Reply:
@Mairramer Does the Swift version still use double-slash? If so, can you add a TODO comment there to fix it?

Reply:
Yes, it continues. Should I assign it to someone?

Reply:
feel free to put my name there